stac_generator/types.py

```diff
@@ -0,0 +1,52 @@
+import datetime
+from collections.abc import Iterable, Mapping, Sequence
+from typing import Literal
+
+import geopandas as gpd
+import pandas as pd
+from httpx._types import PrimitiveData
+
+# Generic Types
+"""Number"""
+NumberT = int | float
+"""Datetime"""
+DateTimeT = datetime.datetime
+"""tuple of start and end dates"""
+TimeExtentT = tuple[DateTimeT | None, DateTimeT | None]  # (start_datetime, end_datetime) format
+
+"""<a href="https://www.rfc-editor.org/rfc/rfc7111">CSV MIME type</a>"""
+CSVMediaType = "text/csv"
+"""<a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types">Excel MIME type</a>"""
+ExcelMediaType = "application/vnd.ms-excel"
+
+"""Pandas DataFrame"""
+PDFrameT = pd.DataFrame
```

Comment:
Some of these types I can get behind but some I dont understand why you would obfuscate the code. Why did you rename the types for the DataFrames in particular? Even the NumberT and DateTimeT I dont really like.

Reply:
It's laziness mainly - i.e. PDFrameT is 2 strokes (P + Tab) whereas pd.DataFrame is 4 strokes (p + d + . + D). Same for NumberT and DateTimeT. I can revert the types no problem.

Reply:
Ok please revert then. I think a few more keypresses are worth the legibility!